packages/component-boilerplate/src/environment/UserIdentityContext.ts

```diff
@@ -187,7 +190,7 @@ export function useUserIdentity(): UserIdentity | undefined {
     | undefined;
 
   const validIdentity = userIdentityFromJwt(identity?.jwt);
-  if (!validIdentity) {
+  if (identity && !validIdentity) {
     // Likely that the JWT has expired
     setUserIdentity(undefined);
```

Comment:
I still like to believe that we don't need this cleanup step, but I'm fine with leaving it in.

```suggestion
    setUserIdentity(undefined);
```